go/vt/vttablet/onlineddl/executor.go

```diff
@@ -1239,6 +1239,45 @@ func (e *Executor) readPendingMigrationsUUIDs(ctx context.Context) (uuids []stri
 	return uuids, err
 }
 
+// readTablePendingMigrationsUUIDs returns UUIDs for migrations on given table, that are in pending state (queued/ready/running)
+func (e *Executor) readTablePendingMigrationsUUIDs(ctx context.Context, table string) (uuids []string, err error) {
+	r, err := e.execQuery(ctx, sqlSelectPendingMigrations)
+	if err != nil {
+		return uuids, err
+	}
+	// we identify running migrations on requested table
+	for _, row := range r.Named().Rows {
+		pendingUUID := row["migration_uuid"].ToString()
+		pendingKeyspace := row["keyspace"].ToString()
+		pendingTable := row["mysql_table"].ToString()
+
+		if pendingKeyspace == e.keyspace && pendingTable == table {
```

Comment:
Here you are filtering in-memory, but for `sqlSelectCompleteMigrationsOnTable` you are filtering in the SQL query. Why the difference? Concerns about inefficient use of the MySQL index if we were to do this one in SQL?

Is there any max size for `_vt.schema_migrations` before we start to garbage collect old rows? Or is it only time based? Or does it grow forever without GC?

Reply:
Good question. There's no particular reasoning. I'm gonna  refactor.